src/ensembl/src/content/app/entity-viewer/shared/helpers/transcripts-sorter.ts

```diff
@@ -53,3 +59,43 @@ export function defaultSort(transcripts: Transcript[]) {
 
   return [...proteinCodingTranscripts, ...sortedNonProteinCodingTranscripts];
 }
+
+export function sortBySplicedLengthLongestToShortest(
+  transcripts: Transcript[]
+) {
+  return [...transcripts].sort((transcriptA, transcriptB) => {
+    return getSplicedRNALength(transcriptB) - getSplicedRNALength(transcriptA);
+  });
+}
+
+export function sortBySplicedLengthShortestToLongest(
+  transcripts: Transcript[]
+) {
+  return [...transcripts].sort((transcriptA, transcriptB) => {
+    return getSplicedRNALength(transcriptA) - getSplicedRNALength(transcriptB);
+  });
+}
+
+export function sortByExonCountHightToLow(transcripts: Transcript[]) {
+  return [...transcripts].sort((transcriptA, transcriptB) => {
+    return transcriptB.spliced_exons.length - transcriptA.spliced_exons.length;
+  });
+}
+
+export function sortByExonCountLowToHigh(transcripts: Transcript[]) {
+  return [...transcripts].sort((transcriptA, transcriptB) => {
+    return transcriptA.spliced_exons.length - transcriptB.spliced_exons.length;
+  });
+}
+
+type SortingFunction = (transcripts: Transcript[]) => Transcript[];
+export const transcriptSortingFunctions: Record<
+  SortingRule,
+  SortingFunction
+> = {
+  default: defaultSort,
+  spliced_length_longest_to_shortest: sortBySplicedLengthLongestToShortest,
+  spliced_length_shortest_to_longest: sortBySplicedLengthShortestToLongest,
+  exon_count_high_to_low: sortByExonCountHightToLow,
+  exon_count_low_to_high: sortByExonCountLowToHigh
```

Comment:
Wouldn't it be simpler to say `spliced_length_desc` & `spliced_length_asc` instead of `spliced_length_longest_to_shortest` & `spliced_length_shortest_to_longest`?

Same for `high_to_low` & `low_to_high`.

Reply:
yeah i think thats good though i prefer longest_to_shortest which is easier to understand. sometimes i have to remind myself what descending and ascending means :) what do you think @azangru ?

And should we also changed sortBySplicedLengthLongestToShortest to sortBySplicedLengthDescending

Reply:
I like asc and desc.

Reply:
@msmanoj @azangru done.